### ISSUE TITLE
Unable to create VM when resource pool contains >=30 VMs

### ISSUE THREAD

**Issue by mesiu84** [hashicorp/terraform-provider-vsphere#1803]
### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

v1.3.6

### Terraform Provider

v2.2.0

### VMware vSphere

7.0.3.00700

### Description

I have 30 VMs in our terraform resource-pool "Terraform" and I can't create another VM, the error that is printed by terraform plan is following

```
Error: error fetching resource pool: path 'Terraform' resolves to multiple resource pools with data.vsphere_resource_pool.pool, on main.tf line 17, in data "vsphere_resource_pool" "pool": 17: data "vsphere_resource_pool" "pool" {
```

If I change resource-pool to one of other pools that I have, where there are less VMs I don't have that problem.

Temporary solution: create empty resource pool

There are no limits set on "Terraform" resource pool, as on other pools.

Same problem exists with Terraform v1.3.0 which was the version that failed first when running terraform plan command

### Affected Resources or Data Sources

Terraform

### Terraform Configuration

```hcl
provider "vsphere" {
  user           = var.vsphere_user
  password       = var.vsphere_password
  vsphere_server = var.vsphere_server
  allow_unverified_ssl = true
}

data "vsphere_datacenter" "dc" {
  name = var.vsphere_datacenter
}

data "vsphere_datastore" "datastore" {
  name          = var.vsphere_datastore
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_resource_pool" "pool" {
  name          = var.vsphere_resource_pool
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_network" "network" {
  name          = var.vsphere_network
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_virtual_machine" "template" {
  name          = var.vsphere_virtual_machine_template
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_tag_category" "category" {
  name = var.vsphere_tag_category
}

data "vsphere_tag" "tag" {
  name        = var.vsphere_tag_name
  category_id = data.vsphere_tag_category.category.id
}

resource "vsphere_virtual_machine" "cloned_virtual_machine" {
  name             = var.vsphere_virtual_machine_name
  resource_pool_id = data.vsphere_resource_pool.pool.id
  datastore_id     = data.vsphere_datastore.datastore.id
  tags             = [data.vsphere_tag.tag.id]

  num_cpus = var.vsphere_cpus
  memory = var.vsphere_memory
  guest_id = data.vsphere_virtual_machine.template.guest_id
  firmware = data.vsphere_virtual_machine.template.firmware
  cpu_hot_add_enabled = true
  cpu_hot_remove_enabled = true
  memory_hot_add_enabled = true

  network_interface {
    network_id   = data.vsphere_network.network.id
    adapter_type = data.vsphere_virtual_machine.template.network_interface_types[0]
  }

  disk {
    label = "disk0"
    size = data.vsphere_virtual_machine.template.disks.0.size
    thin_provisioned = data.vsphere_virtual_machine.template.disks.0.thin_provisioned
  }
  disk {
    label = "disk1"
    size = data.vsphere_virtual_machine.template.disks.1.size
    thin_provisioned = data.vsphere_virtual_machine.template.disks.1.thin_provisioned
    unit_number = 1
  }
  disk {
    label = "disk2"
    size = data.vsphere_virtual_machine.template.disks.2.size
    thin_provisioned = data.vsphere_virtual_machine.template.disks.2.thin_provisioned
    unit_number = 2
  }

  clone {
    template_uuid = data.vsphere_virtual_machine.template.id
    timeout = 30
    customize {
      linux_options {
        host_name = var.vsphere_virtual_machine_hostname
        domain    = var.vsphere_virtual_machine_domain
      }
      network_interface {
        ipv4_address       = var.vsphere_private_ip_address
        ipv4_netmask       = var.vsphere_netmask
      }
      dns_server_list    = ["8.8.8.8", "8.8.4.4"]
      ipv4_gateway       = var.vsphere_gateway
      timeout = 30
    }
  }
}
```

### Debug Output

```
data.vsphere_datacenter.dc: Reading...
data.vsphere_tag_category.category: Reading...
data.vsphere_datacenter.dc: Read complete after 0s [id=datacenter-2]
data.vsphere_network.network: Reading...
data.vsphere_virtual_machine.template: Reading...
data.vsphere_datastore.datastore: Reading...
data.vsphere_resource_pool.pool: Reading...
data.vsphere_tag_category.category: Read complete after 1s [id=urn:vmomi:InventoryServiceCategory:XXXXXXXXXXX:GLOBAL]
data.vsphere_tag.tag: Reading...
data.vsphere_network.network: Read complete after 2s [id=dvportgroup-438026]
data.vsphere_datastore.datastore: Read complete after 3s [id=datastore-233728]
data.vsphere_virtual_machine.template: Read complete after 4s [id=XXXXXXXXXXXXXXXXX]
data.vsphere_tag.tag: Read complete after 4s [id=urn:vmomi:InventoryServiceTag:XXXXXXXXXXXXXXXX:GLOBAL]
data.vsphere_resource_pool.pool: Still reading... [10s elapsed]
╷
│ Error: error fetching resource pool: path 'Terraform' resolves to multiple resource pools
│ 
│   with data.vsphere_resource_pool.pool,
│   on main.tf line 17, in data "vsphere_resource_pool" "pool":
│   17: data "vsphere_resource_pool" "pool" {

```

### Expected Behavior

terraform plan doesn't fail

### Actual Behavior

Terraform plan fails because of "high" (30) number of VMs 
Doesn't fail when running on resources with less VMs (12, 16)
I only have 1 Pool where there is 30 VMs and that one is used by terraform

### Steps to Reproduce

Use on resource pool where there are 30 VMs running already

**Comment by dimitarproynov** [comment]
Hi @mesiu84 ,

I've cloned 30 VMs in a ResourcePool and powered them on using the vSphere provider. I've had no issues when I've added another clone, bringing the count up to 31.

To me it seems that somebody created another ResourcePool, named "Terraform" in your environment, causing the error you're observing.

Can you try to rename your ResourcePool with 30 VMs from "Terraform" to "Terraform1", change your var.vsphere_resource_pool to address this and rerun "terraform plan"?

Regards,
_Proynov

**Comment by mesiu84** [comment]
You might be right, just noticed that someone created another pool with the same name but in different clusters, looks like someone did there some cleaning by moving few VMs between those and now terraform is failing because of that. Thank you for a good hint, I didn't thought about it

**Comment by dimitarproynov** [comment]
Have you tried renaming your RP to "Terraform1" and retrying "terraform plan"?

**Comment by mesiu84** [comment]
Not yet, but will try it today after meetings

**Comment by mesiu84** [comment]
I renamed one of those pools to Terraform1 and it started working again. The question is now, could we add another parameter which would identify cluster name in which that pool exists?
As I wrote it turned out there are 2 pools with the same name "Terraform" but each one is in different Cluster (second was created after new Cluster was added few weeks ago), right now we don't have any parameter in our terraform files to identify which Cluster it should be so terraform searches in both and fails because both have a pool with the same name.
